vignettes/sensors.Rmd

```diff
@@ -16,138 +16,163 @@ knitr::opts_chunk$set(
+ggplot(a, aes(x = t, y= value)) + geom_line() + 
+  facet_wrap(~axis, dir = 'v', strip.position = 'right') + ylim(c(-0.025, 0.025))
 ```
+We are looking at 10s of data sampled at a high enough sampling rate (for accelerometer data) of 100Hz. The data looks noisy, and looks like it could make good use of some pre-processing like frequency filtering, maybe subsetting the time (time filtering) -- all of which `mhealthtools` conveniently offers.
 
 #### Extracting features (using pre-existing pipeline)
-To use the existing feature set (you can define your own features or feed in your own model to extract features in `mhealthtools`!!). Note: You need to get the data in the proper format (as that of the sample data) to use the feature extraction pipeline.
 
-```{r, warning=FALSE, error=FALSE, prompt=FALSE}
+Let's take a look at the default feature extraction pipeline, whose features are all classical signal processing features based on time domain, frequency domain and the energy bands of the frequency spectrum.
+
+```{r accel default features, warning=FALSE, error=FALSE, prompt=FALSE}
 accel_features <- accelerometer_features(accelerometer_data)
 # using default parameters
 ```
+The output file is a list containing features extracted using default functions (`$extracted_features`), features calculated using models (`$model_features`) and an element for error (`$error`). Let's look at the `$extracted_features` to see how the default features look like
 
-```{r}
-dplyr::glimpse(accel_features)
+```{r show accel default features}
+dplyr::glimpse(accel_features$extracted_features)
 ```
 
-`accelerometer_features` also offers multiple parameter options. The follwing examples are an effort to make the user aware of the options. 
+`accelerometer_features` also offers multiple parameter options. The following examples are an effort to make the user aware of the options. 
 
-In the following example, we change the window length (to 128, with 256 being the default), having the same overlap for the same `accel_data` used before
+In the following example, we change the window length (to 128, with NULL being the default - i.e no windowing), having the 50% overlap between consecutive windows for the same `accelerometer_data` used before
 
 ```{r, warning=FALSE, error=FALSE, prompt=FALSE}
 accel_features <- accelerometer_features(
   accelerometer_data,
   window_length = 128,
   window_overlap = 0.5)
-# Consider a window length of 128 samples
+# Consider a window length of 128 samples,
+# 50% overlap between consecutive windows
 ```
 
 ```{r}
-dplyr::glimpse(accel_features)
+dplyr::glimpse(accel_features$extracted_features)
 ```
+Notice how we now have three extra columns in the feature extraction output namely, `window`, `window_start_time` and `window_stop_time` - which indicate the number of the window in the signal (the first window is window 1, the next consecutive window is window 2 and so on), and the start and stop time respectively of that window.
```

Comment:
- which indicate the number of the **windows** in the signal

Reply:
Changing to "number of each window..."